homeassistant/components/fibaro/quality_scale.yaml

```diff
@@ -0,0 +1,77 @@
+rules:
```

Comment:
So what is `FibaroController` and why should it belong in HA as opposed to be part of the client library? Because it feels very device/service specific

Reply:
FibaroController is a push based counterpart of the DataUpdateCoordinator.
Question: is the DataUpdateCoordinator also usable for push based integrations?

Additionally it translates the data structures which are provided by fibaro to the data structures required by HA. As this logic is more HA specific than API specific it is currently put in HA code base. If we move this to pyfibaro library we need to add a third data structure which matches the data structure of HA but is unrelated to it. Also in zwave_js integration the discovery mechanism of the devices is in the HA repository, so it should be good here.

Reply:
We do have moments where a coordinator is used for push based data. I think watergate is one that comes to mind which does that I believe. It would only be useful if you have a full state for all devices or a single devices, not just partial updates.

Reply:
The fibaro just sends partial updates. So using the FibaroController as it is implemented currently is ok?

Reply:
Would it make sense to just subscribe to them on entity level then? If we can't piece together a state, the coordinator is not really useful for this.

Reply:
Coordinator is anyway needed to route the state updates to the entities, there is just one subscription on fibaro hub which returns all state updates of all devices. But as stated in the other https://github.com/home-assistant/core/pull/133279#discussion_r1889183475 it could be beneficial to optimize the data transfer to the entities.

Reply:
> Coordinator is anyway needed to route the state updates to the entities,

How come? Would it be able to update the library to register multiple listeners with the identifier of the device? And when you get an event you call all the listeners. This way all the relevant entities are updated without the need of a coordinator. WDYT?

Other option would be using dispatchers and a centralised callback